src/nni_manager/core/nniDataStore.ts

```diff
@@ -161,11 +167,31 @@ class NNIDataStore implements DataStore {
 
     private async getFinalMetricData(trialJobId: string): Promise<any> {
         const metrics: MetricDataRecord[] = await this.getMetricData(trialJobId, 'FINAL');
-        assert(metrics.length <= 1);
-        if (metrics.length === 1) {
+
+        const multiPhase: boolean | undefined = await this.isMultiPhase();
+
+        if (multiPhase) {
+            if (metrics.length > 0) {
+                return metrics[metrics.length - 1];
+            } else {
+                return undefined;
+            }
+        } else {
+            assert(metrics.length <= 1);
+
             return metrics[0];
+        }
+    }
+
+    private async isMultiPhase(): Promise<boolean|undefined> {
+        if (this.multiPhase === undefined) {
+            this.multiPhase = (await this.getExperimentProfile(getExperimentId())).params.multiPhase;
+        }
+
+        if (this.multiPhase !== undefined) {
```

Comment:
Optional: change to 
if (this.multiPhase) { 
}
for simplicity

Reply:
Fixed.